torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4334,7 +4334,13 @@ def make_idx(n):
     else:
         alphas = (None,)
 
-    for shape, alpha in product(shapes, alphas):
+    if fill:
+        # A weird number to catch errors
+        values = (make_arg((1,)).item(), make_arg((1,)).squeeze(0))
```

Comment:
For second `value` can you mention that first value is for `index_fill.int_Scalar` overload and second for `index_fill.int_Tensor` overload.

Also, `make_arg(())` should give you a scalar tensor.

Reply:
Sure.

The reason I used `make_arg((1,)).squeeze(0)` was because it caught more errors than simply using `make_arg(())`.

Reply:
Both should return scalar tensors. 

Can you check what are the errors that `make_arg((1,)).squeeze(0)` causes? (I don't think it should lead to different errors 🤔 )

Ref:
```python
import torch

x = torch.testing.make_tensor((), device='cpu', dtype=torch.float)
y = torch.testing.make_tensor((1,), device='cpu', dtype=torch.float).squeeze(0)
print(x.shape, y.shape)  # prints "torch.Size([]) torch.Size([])"
```

Reply:
@kshitij12345 These tests fail with `make_arg((1,)).squeeze(0)` only:
```
_______________________________________________________ TestCommonCPU.test_noncontiguous_samples_index_fill_cpu_complex64 _______________________________________________________
Traceback (most recent call last):
  File "/Users/qqaatw/Forks/pytorch/test/test_ops.py", line 655, in test_noncontiguous_samples
    self.assertEqual(t, n, msg=msg.format(i))
  File "/Users/qqaatw/Forks/pytorch/torch/testing/_internal/common_utils.py", line 2966, in assertEqual
    raise error_metas[0].to_error(
AssertionError: None mismatch: None is not (0.05548311024904251+0.5575037002563477j)
Got different gradients for contiguous / non-contiguous inputs wrt input 1.
________________________________________________________ TestCommonCPU.test_noncontiguous_samples_index_fill_cpu_float32 ________________________________________________________
Traceback (most recent call last):
  File "/Users/qqaatw/Forks/pytorch/test/test_ops.py", line 655, in test_noncontiguous_samples
    self.assertEqual(t, n, msg=msg.format(i))
  File "/Users/qqaatw/Forks/pytorch/torch/testing/_internal/common_utils.py", line 2966, in assertEqual
    raise error_metas[0].to_error(
AssertionError: None mismatch: None is not 0.2167382389307022
Got different gradients for contiguous / non-contiguous inputs wrt input 1.
________________________________________________________ TestCompositeComplianceCPU.test_backward_index_fill_cpu_float32 ________________________________________________________
Traceback (most recent call last):
  File "/Users/qqaatw/Forks/pytorch/test/test_ops.py", line 1446, in test_backward
    composite_compliance.check_backward_formula(
  File "/Users/qqaatw/Forks/pytorch/torch/testing/_internal/composite_compliance.py", line 491, in check_backward_formula
    assert_equal_fn(tuple(map(unwrap, actual)), expected, equal_nan=True)
  File "/Users/qqaatw/Forks/pytorch/torch/testing/_internal/common_utils.py", line 2966, in assertEqual
    raise error_metas[0].to_error(
AssertionError: None mismatch: 1.0 is not None

The failure occurred for item [1]
```

Reply:
Please look into those, both incantations should produce the same sample arguments, please see what's going differently with those.

Reply:
`make_arg((1,)).squeeze(0)` means that the sample input isn't a leaf node in autograd, therefore `input.grad` is `None` which is what most of those errors are complaining about.

Reply:
@peterbell10 Thank you for shedding light on it! I missed this part, thinking why the grad was None for a long while as the graph was connected without any part being detached.